MobileCenterPush/MobileCenterPush/MSPush.m

```diff
@@ -0,0 +1,277 @@
+#import "MSPush.h"
+#import "MSPushLog.h"
+#import "MSPushPrivate.h"
+#import "MSPushInternal.h"
+#import "MSDeviceTracker.h"
+#import "MSMobileCenterInternal.h"
+
+#if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0
+
+#import <UserNotifications/UserNotifications.h>
+
+#endif
+
+/**
+ * Service storage key name.
+ */
+static NSString *const kMSServiceName = @"Push";
+
+/**
+ * Key for storing push token
+ */
+static NSString *const kMSPushServiceStorageKey = @"kmspushservicepushstoringkey";
+
+/**
+ * Singleton
+ */
+static MSPush *sharedInstance = nil;
+static dispatch_once_t onceToken;
+
+@implementation MSPush
+
+@synthesize deviceTokenHasBeenSent;
+@synthesize isRequestInProgress;
+
+#pragma mark - Service initialization
+
+- (instancetype)init {
+
+  self = [super init];
+
+  if (self) {
+
+    deviceTokenHasBeenSent = NO;
+    isRequestInProgress = NO;
+  }
+
+  return self;
+}
+
+#pragma mark - MSServiceInternal
+
++ (instancetype)sharedInstance {
+  dispatch_once(&onceToken, ^{
+    if (sharedInstance == nil) {
+      sharedInstance = [self new];
+    }
+  });
+  return sharedInstance;
+}
+
+- (void)startWithLogManager:(id<MSLogManager>)logManager appSecret:(NSString *)appSecret {
+  [super startWithLogManager:logManager appSecret:appSecret];
+
+  MSLogVerbose([MSPush logTag], @"Started push service.");
+
+  if( [MSPush isEnabled] ) {
+
+    [self registerForRemoteNotifications];
+  }
+}
+
++ (NSString *)logTag {
+  return @"MobileCenterPush";
+}
+
+- (NSString *)storageKey {
+  return kMSServiceName;
+}
+
+- (MSPriority)priority {
+  return MSPriorityDefault;
+}
+
+- (MSInitializationPriority)initializationPriority {
```

Comment:
Can you add a TODO comment and say that this method should be removed after merging `base/updates`?
We have another working branch that will be merged to `develop` in March, this doesn't required here once we have that. Just add TODO here so that we don't forget about it.